### ISSUE TITLE
Prepare project for eslint-config-airbnb v12.x

### ISSUE THREAD

**Issue by gihrig** [mxstbr/react-boilerplate#1029]
eslint-config-airbnb just released v12.0.0 This release triggers several new linting errors that will need to be resolved before v12.x can be adopted.

It would be awesome if someone could work through these issues and open a PR.

**"eslint-config-airbnb":    11.2.0  →   12.0.0

```bash
~/react-boilerplate/app/app.js
  15:8  error  Missing file extension for "file?name=[name].[ext]!./.htaccess"  import/extensions

~/react-boilerplate/internals/templates/app.js
  12:8  error  Missing file extension for "file?name=[name].[ext]!./.htaccess"  import/extensions

~/react-boilerplate/internals/templates/asyncInjectors.test.js
  6:28  error  Unexpected use of file extension "js" for "store.js"  import/extensions

~/react-boilerplate/internals/webpack/webpack.dev.babel.js
   11:13  error  Calls to require() should use string literals  import/no-dynamic-require
   96:19  error  Calls to require() should use string literals  import/no-dynamic-require
  117:17  error  Calls to require() should use string literals  import/no-dynamic-require

~/react-boilerplate/internals/webpack/webpack.dll.babel.js
  14:13  error  Calls to require() should use string literals  import/no-dynamic-require

~/react-boilerplate/server/middlewares/frontendMiddleware.js
  5:13  error  Calls to require() should use string literals  import/no-dynamic-require
```

**Comment by amilajack** [comment]
`import/no-dynamic-require` seems like the kind of issue that would have to be disabled a lot.

**Comment by gihrig** [comment]
@mxstbr `import/no-dynamic-require` seems to be a toss up between static analysis tools and developer convenience. What's your preference: Local or Global disabling of the rule?

Rule proposal [eslint-plugin-import#567](https://github.com/benmosher/eslint-plugin-import/issues/567)
Documentation [eslint-plugin-import docs](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-dynamic-require.md)

**Comment by mxstbr** [comment]
I think we should disable it for now, I don't have the time to look into it though. If you think there's a better way of doing it, let me know!

**Comment by amilajack** [comment]
I can make a PR for this. I think we should run the linter on as many files we can. Thoughts on this?

**Comment by mxstbr** [comment]
> I can make a PR for this. I think we should run the linter on as many files we can.

Yes please!

**Comment by gihrig** [comment]
@amilajack PR?

**Comment by amilajack** [comment]
Sure thing!

**Comment by amilajack** [comment]
Question: What is the `store.js` dependency [here](https://github.com/mxstbr/react-boilerplate/blob/master/internals/templates/asyncInjectors.test.js#L6)? From what I gathered, it seems like a template is being copied into it. Is this correct?

**Comment by gihrig** [comment]
@justingreenberg should be able to answer that

**Comment by justingreenberg** [comment]
@gihrig @amilajack it is just the store config... those are core templates used when running `clean` https://github.com/mxstbr/react-boilerplate/blob/master/internals/scripts/clean.js#L82

we should prob disable import linting for those files... they are not meant to be run, so many of the relative imports are broken (our `internal` directory structure does not match generated output)

**Comment by amilajack** [comment]
@justingreenberg is the `.js` suffix necessary? Is it being imported as a local file?

**Comment by justingreenberg** [comment]
@amilajack in the test file, the relative import should really be '../store' since destination folder for test file is `/tests/` (however it still works in new projects due to webpack resolving from project root, where store is located)

**Comment by gihrig** [comment]
Ref PR #1067

**Comment by gihrig** [comment]
> our internal directory structure does not match generated output

#865 was opened to address that. It was prematurely closed (by me) because the error was made to go away. But the core problem still exists. Something else that would like attention...

**Comment by gihrig** [comment]
Closed as result of #1067.
